lib/api/core/plugins/pluginsManager.js

```diff
@@ -243,6 +243,30 @@ class PluginsManager {
   listStrategies () {
     return Bluebird.resolve(this.registeredStrategies);
   }
+
+  /**
+   * Stops and unregisters plugins workers
+   * @returns {Promise}
+   */
+  shutdownWorkers() {
+    const names = Object.keys(this.workers);
+
+    if (names.length === 0) {
+      return Bluebird.resolve();
+    }
+
+    return new Bluebird(resolve => {
+      names.forEach(name => {
+        this.workers[name].pmIds.getArray().forEach(id => pm2.delete(id));
+      });
+
+      this.workers = {};
+
+      // gives time for pm2 to delete plugins before resolving
+      // the promise
+      setTimeout(() => resolve(), 100);
```

Comment:
I didn't know about Bluebird.delay, I like that :-)

About plugins, I didn't thought about that. Normally, once PM2 has received the `delete` command, it waits a small delay (1.6s by default) for the process to stop, and it sends a SIGKILL after that delay.
Buuuut... we may have to change that default delay. And waiting for plugins to stop and free allocated resource before creating new instances seems a cleaner to deal with plugins to me.

I'll fix that, and add the appropriate tests.